spark/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/deltaMerge.scala

```diff
@@ -479,12 +485,15 @@ object DeltaMergeInto {
             // column name. The target columns do not need resolution. The right hand side
             // expression (i.e. sourceColumnBySameName) needs to be resolved only by the source
             // plan.
-            fakeTargetPlan.output.map(_.name).map { tgtColName =>
-              val resolvedExpr = resolveOrFail(
```

Comment:
So, triggering resolution for each column -- when resolution already pays `O(#columns)` work -- meant quadratic work overall? Pernicious!